addons/website_sale_loyalty/models/loyalty.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, tools
+from odoo.tools.safe_eval import safe_eval
+
+
+class LoyaltyProgram(models.Model):
+    _name = 'loyalty.program'
+    _inherit = ['loyalty.program']
+
+    def get_available_rewards(self, order):
+        """ Returns the list of rewards that the cart can get """
+        client = order.partner_id
+        rewards = []
+        if client:
+            spendable_points = tools.float_round(client.loyalty_points - order.spent_loyalty_points, 0, rounding_method='HALF-UP')
+            for reward in self.reward_ids:
+                if reward.minimum_points > spendable_points:
+                    continue
+                elif reward.reward_type == 'discount' and reward.point_cost > spendable_points:
+                    continue
+                elif reward.reward_type == 'gift' and reward.point_cost > spendable_points:
+                    continue
+                elif reward.reward_type == 'discount' and reward.discount_apply_on == 'specific_products':
+                    if not (order.order_line.mapped('product_id') & reward.discount_specific_product_ids):
```

Comment:
Since Odoo 13.0, `mapped` is now built-in for relational fields:
```suggestion
                    if not (order.order_line.product_id & reward.discount_specific_product_ids):
```